### PR TITLE
First pass implementation of chords

### PR DESCRIPTION
**Pull request by jystic**
The most important part of this is that it gets the structure of the control flow right for doing chords.

The `Entity -> Set DateTime` lookup function is really stupid/slow but this can be improved in a later PR.

### CODE REVIEW

File: src/Icicle/Sea/FromAvalanche/Psv.hs [hunk 1]
```diff
@@ -820,6 +960,9 @@ lookupTombstones :: PsvConfig -> SeaProgramState -> Set Text
 lookupTombstones config state =
   fromMaybe Set.empty (Map.lookup (stateAttribute state) (psvTombstones config))
 
+------------------------------------------------------------------------
+-- Should be in P?
+
 last :: [a] -> Maybe a
 last []     = Nothing
 last (x:[]) = Just x
```
Comment (tranma) [inline src/Icicle/Sea/FromAvalanche/Psv.hs]: http://hackage.haskell.org/package/base-4.8.1.0/docs/Data-Maybe.html#v:listToMaybe?

**Comment by tranma** [comment]
the C looks fine :+1:

File: src/Icicle/Sea/FromAvalanche/Psv.hs (hunk 1 above)
Comment (jystic) [inline src/Icicle/Sea/FromAvalanche/Psv.hs]: lol

Comment (jystic) [inline src/Icicle/Sea/FromAvalanche/Psv.hs]: oh wait, not lol, `listToMaybe` is `head` not `last`

Comment (tranma) [inline src/Icicle/Sea/FromAvalanche/Psv.hs]: ok, this `last` is not exhaustive though

Comment (tranma) [inline src/Icicle/Sea/FromAvalanche/Psv.hs]: `listToMaybe . reverse` 😂😂😂

Comment (amosr) [inline src/Icicle/Sea/FromAvalanche/Psv.hs]: oh, github just cut off the last line

File: data/sea/50-psv.h [hunk 2]
```diff
@@ -355,18 +355,19 @@ static psv_error_t psv_read_buffer (psv_state_t *s)
         const char  *value_end  = time_ptr - 1;
         const size_t value_size = value_end - value_ptr;
 
-        const bool new_entity = entity_cur_size != entity_size
-                             || memcmp (entity_cur, entity_ptr, entity_size) != 0;
+        const bool new_entity = memcmp (entity_cur, entity_ptr, entity_size) != 0;
```
Comment (amosr) [inline data/sea/50-psv.h]: ah, okay. because entity_size *includes* the pipe, the size comparison is unnecessary?

**Comment by amosr** [comment]
:+1:

File: data/sea/50-psv.h (hunk 2 above)
Comment (jystic) [inline data/sea/50-psv.h]: yep, but as discussed IRL, i might need to roll this back :(

**Pull request merged**